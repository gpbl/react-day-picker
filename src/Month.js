import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Weekdays from './Weekdays';
import Day from './Day';
import { ENTER } from './keys';

import * as ModifiersUtils from './ModifiersUtils';
import * as Helpers from './Helpers';
import * as DateUtils from './DateUtils';

export default class Month extends Component {
  static propTypes = {
    classNames: PropTypes.shape({
      body: PropTypes.string.isRequired,
      month: PropTypes.string.isRequired,
      outside: PropTypes.string.isRequired,
      today: PropTypes.string.isRequired,
      week: PropTypes.string.isRequired,
      weekNumber: PropTypes.string.isRequired,
      disabled: PropTypes.string.isRequired,
      selected: PropTypes.string.isRequired,
    }).isRequired,
    tabIndex: PropTypes.number,

    month: PropTypes.instanceOf(Date).isRequired,
    months: PropTypes.arrayOf(PropTypes.string),

    modifiersStyles: PropTypes.object,

    showWeekDays: PropTypes.bool,
    showOutsideDays: PropTypes.bool,

    renderDay: PropTypes.func.isRequired,
    renderWeek: PropTypes.func.isRequired,

    captionElement: PropTypes.oneOfType([
      PropTypes.element,
      PropTypes.func,
      PropTypes.instanceOf(React.Component),
    ]).isRequired,
    weekdayElement: PropTypes.oneOfType([
      PropTypes.element,
      PropTypes.func,
      PropTypes.instanceOf(React.Component),
    ]),

    fixedWeeks: PropTypes.bool,
    showWeekNumbers: PropTypes.bool,

    locale: PropTypes.string.isRequired,
    localeUtils: PropTypes.object.isRequired,
    weekdaysLong: PropTypes.arrayOf(PropTypes.string),
    weekdaysShort: PropTypes.arrayOf(PropTypes.string),
    firstDayOfWeek: PropTypes.number.isRequired,

    onCaptionClick: PropTypes.func,
    onDayClick: PropTypes.func,
    onDayFocus: PropTypes.func,
    onDayKeyDown: PropTypes.func,
    onDayMouseEnter: PropTypes.func,
    onDayMouseLeave: PropTypes.func,
    onDayMouseDown: PropTypes.func,
    onDayMouseUp: PropTypes.func,
    onDayTouchEnd: PropTypes.func,
    onDayTouchStart: PropTypes.func,
    onWeekClick: PropTypes.func,
  };

  renderDay = day => {
    const monthNumber = this.props.month.getMonth();
    const propModifiers = Helpers.getModifiersFromProps(this.props);
    const dayModifiers = ModifiersUtils.getModifiersForDay(day, propModifiers);
    if (
      DateUtils.isSameDay(day, new Date()) &&
      !Object.prototype.hasOwnProperty.call(
        propModifiers,
        this.props.classNames.today
      )
    ) {
      dayModifiers.push(this.props.classNames.today);
    }
    if (day.getMonth() !== monthNumber) {
      dayModifiers.push(this.props.classNames.outside);
    }

    const isOutside = day.getMonth() !== monthNumber;
    let tabIndex = -1;
    // Focus on the first day of the month
    if (this.props.onDayClick && !isOutside && day.getDate() === 1) {
      tabIndex = this.props.tabIndex; // eslint-disable-line prefer-destructuring
    }
    const key = `${day.getFullYear()}${day.getMonth()}${day.getDate()}`;
    const modifiers = {};
    dayModifiers.forEach(modifier => {
      modifiers[modifier] = true;
    });

    return (
      <Day
        key={`${isOutside ? 'outside-' : ''}${key}`}
        classNames={this.props.classNames}
        day={day}
        modifiers={modifiers}
        modifiersStyles={this.props.modifiersStyles}
        empty={
          isOutside && !this.props.showOutsideDays && !this.props.fixedWeeks
        }
        tabIndex={tabIndex}
        ariaLabel={this.props.localeUtils.formatDay(day, this.props.locale)}
        ariaDisabled={
          isOutside || dayModifiers.indexOf(this.props.classNames.disabled) > -1
        }
        ariaSelected={dayModifiers.indexOf(this.props.classNames.selected) > -1}
        onClick={this.props.onDayClick}
        onFocus={this.props.onDayFocus}
        onKeyDown={this.props.onDayKeyDown}
        onMouseEnter={this.props.onDayMouseEnter}
        onMouseLeave={this.props.onDayMouseLeave}
        onMouseDown={this.props.onDayMouseDown}
        onMouseUp={this.props.onDayMouseUp}
        onTouchEnd={this.props.onDayTouchEnd}
        onTouchStart={this.props.onDayTouchStart}
      >
        {this.props.renderDay(day, modifiers)}
      </Day>
    );
  };

  render() {
    const {
      classNames,

      month,
      months,

      fixedWeeks,
      captionElement,
      weekdayElement,

      locale,
      localeUtils,
      weekdaysLong,
      weekdaysShort,
      firstDayOfWeek,

      onCaptionClick,

      showWeekNumbers,
      showWeekDays,
      onWeekClick,
    } = this.props;

    const captionProps = {
      date: month,
      classNames,
      months,
      localeUtils,
      locale,
      onClick: onCaptionClick ? e => onCaptionClick(month, e) : undefined,
    };
    const caption = React.isValidElement(captionElement)
      ? React.cloneElement(captionElement, captionProps)
      : React.createElement(captionElement, captionProps);

    const weeks = Helpers.getWeekArray(month, firstDayOfWeek, fixedWeeks);
    return (
      <div className={classNames.month} role="grid">
        {caption}
        {showWeekDays && (
          <Weekdays
            classNames={classNames}
            weekdaysShort={weekdaysShort}
            weekdaysLong={weekdaysLong}
            firstDayOfWeek={firstDayOfWeek}
            showWeekNumbers={showWeekNumbers}
            locale={locale}
            localeUtils={localeUtils}
            weekdayElement={weekdayElement}
          />
        )}
        <div className={classNames.body} role="rowgroup">
          {weeks.map(week => {
            let weekNumber;
            if (showWeekNumbers) {
              weekNumber = DateUtils.getWeekNumber(week[6]);
            }
            return (
              <div
                key={week[0].getTime()}
                className={classNames.week}
                role="row"
              >
                {showWeekNumbers && (
                  <div
                    className={classNames.weekNumber}
                    tabIndex={onWeekClick ? 0 : -1}
                    role="gridcell"
                    onClick={
                      onWeekClick
                        ? e => onWeekClick(weekNumber, week, e)
                        : undefined
                    }
                    onKeyUp={
                      onWeekClick
                        ? e =>
                            e.keyCode === ENTER &&
                            onWeekClick(weekNumber, week, e)
                        : undefined
                    }
                  >
                    {this.props.renderWeek(weekNumber, week, month)}
                  </div>
                )}
                {week.map(this.renderDay)}
              </div>
            );
          })}
        </div>
      </div>
    );
  }
}
