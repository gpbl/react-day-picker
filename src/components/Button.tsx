import React from "react";

/**
 * Render the button elements in the calendar.
 *
 * @group Components
 * @see https://daypicker.dev/guides/custom-components
 */
export function Button(props: JSX.IntrinsicElements["button"]) {
  return <button {...props} />;
}

export type ButtonProps = Parameters<typeof Button>[0];
