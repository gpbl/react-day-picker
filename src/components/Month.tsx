import React from "react";

import type { CalendarMonth } from "../classes/CalendarMonth.js";

/**
 * Render the grid with the weekday header row and the weeks for the given
 * month.
 *
 * @group Components
 * @see https://daypicker.dev/guides/custom-components
 */
export function Month(
  props: {
    /** The month where the grid is displayed. */
    calendarMonth: CalendarMonth;
    /** The index where this month is displayed. */
    displayIndex: number;
  } & JSX.IntrinsicElements["div"]
) {
  const { calendarMonth, displayIndex, ...divProps } = props;
  return <div {...divProps}>{props.children}</div>;
}

export type MonthProps = Parameters<typeof Month>[0];
