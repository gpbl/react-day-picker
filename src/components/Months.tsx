import React from "react";

/**
 * Component wrapping the month grids.
 *
 * @group Components
 * @see https://daypicker.dev/guides/custom-components
 */
export function Months(props: JSX.IntrinsicElements["div"]) {
  return <div {...props} />;
}

export type MonthsProps = Parameters<typeof Months>[0];
