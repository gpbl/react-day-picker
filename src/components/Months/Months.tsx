import { PropsWithChildren } from 'react';

import { useDayPicker } from 'contexts/DayPicker';

/** The props for the {@link Months} component. */
export type MonthsProps = PropsWithChildren;

/**
 * Render the wrapper for the mont grids.
 */
export function Months(props: MonthsProps): JSX.Element {
  const { classNames, styles } = useDayPicker();

  return (
    <div className={classNames.months} style={styles.months}>
      {props.children}
    </div>
  );
}
