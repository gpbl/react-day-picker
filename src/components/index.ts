/* These exported components will be customizable with the `components` prop. */
export * from './Calendar';
export * from './DayGridCell';
export * from './Footer';
export * from './Months';
export * from './MonthCaption';
export * from './MonthGrid';
export * from './Nav';
export * from './WeekdayColumnHeader';
export * from './WeekdaysRow';
export * from './WeekNumberRowHeader';
export * from './WeekRow';
export * from './deprecated';
