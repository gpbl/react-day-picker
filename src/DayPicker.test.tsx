import React from "react";

import { fireEvent, render, screen } from "@testing-library/react";
import { startOfDay, startOfMonth } from "date-fns";

import {
  activeElement,
  dateButton,
  grid,
  nav,
  previousButton
} from "@/test/elements";
import { user } from "@/test/user";

import { DayPicker } from "./DayPicker";
import { MonthProps } from "./components/Month";
import { MonthsProps } from "./components/Months";

const testId = "test";
const dayPicker = () => screen.getByTestId(testId);

test("should render a date picker component", () => {
  render(<DayPicker data-testid={testId} />);
  expect(dayPicker()).toBeInTheDocument();
});

it("should render the navigation and month grids", () => {
  render(<DayPicker data-testid={testId} />);

  expect(nav()).toBeInTheDocument();
  expect(grid()).toBeInTheDocument();
});

it("should apply classnames and style according to props", () => {
  render(
    <DayPicker
      data-testid={testId}
      className="custom-class"
      numberOfMonths={2}
      showWeekNumber
      style={{ color: "red" }}
    />
  );

  expect(dayPicker()).toHaveClass("rdp-root");
  expect(dayPicker()).toHaveClass("custom-class");
  expect(dayPicker()).toHaveStyle({ color: "red" });
});

it("should use custom components", () => {
  render(
    <DayPicker
      data-testid={testId}
      components={{
        Nav: () => <div>Custom Navigation</div>,
        Month: (props: MonthProps) => <div>Custom Month</div>,
        Months: (props: MonthsProps) => (
          <div {...props}>
            Custom Months<div>{props.children}</div>
          </div>
        ),
        Footer: () => <div>Custom Footer</div>
      }}
      footer="Footer"
    />
  );

  expect(dayPicker()).toHaveTextContent("Custom Navigation");
  expect(dayPicker()).toHaveTextContent("Custom Months");
  expect(dayPicker()).toHaveTextContent("Custom Month");
  expect(dayPicker()).toHaveTextContent("Custom Footer");
});

describe("when the date picker is focused", () => {
  test("focus the previous button", async () => {
    render(<DayPicker />);
    await user.tab();
    expect(activeElement()).toBe(previousButton());
  });

  test("on RTL, focus the previous button", async () => {
    render(<DayPicker dir="rtl" />);
    await user.tab();
    expect(activeElement()).toBe(previousButton());
  });
});

describe("when the grid is focused", () => {
  const today = new Date();

  beforeAll(() => jest.setSystemTime(today));
  afterAll(() => jest.useRealTimers());

  test("should focus the today's date", async () => {
    render(<DayPicker mode="single" today={today} />);
    await user.tab();
    await user.tab();
    await user.tab();
    expect(activeElement()).toBe(dateButton(today));
  });
  describe("when the todayâ€™s date is disabled", () => {
    test("should focus the first day of the month", async () => {
      render(<DayPicker mode="single" disabled={today} />);
      await user.tab();
      await user.tab();
      await user.tab();
      expect(activeElement()).toBe(dateButton(startOfMonth(today)));
    });
  });
});

describe("when a day is mouse entered", () => {
  const handleDayMouseEnter = jest.fn();
  const handleDayMouseLeave = jest.fn();
  const today = startOfDay(new Date());
  beforeEach(async () => {
    render(
      <DayPicker
        today={today}
        defaultMonth={today}
        mode="single"
        onDayMouseEnter={handleDayMouseEnter}
        onDayMouseLeave={handleDayMouseLeave}
      />
    );
    fireEvent.mouseEnter(dateButton(today));
    fireEvent.mouseLeave(dateButton(today));
  });
  test("should call the event handler", async () => {
    expect(handleDayMouseEnter).toHaveBeenCalled();
    expect(handleDayMouseLeave).toHaveBeenCalled();
  });
});
