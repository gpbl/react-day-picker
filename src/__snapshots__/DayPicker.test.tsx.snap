// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should localize the calendar should use the custom locale 1`] = `
<body>
  <div>
    <div
      class="rdp-root"
    >
      <div
        class="rdp-months"
      >
        <nav
          aria-label=""
          class="rdp-nav"
        >
          <button
            aria-label="Go to the Previous Month"
            class="rdp-button_previous"
            type="button"
          >
            <svg
              class="rdp-chevron"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <polygon
                points="16 18.112 9.81111111 12 16 5.87733333 14.0888889 4 6 12 14.0888889 20"
              />
            </svg>
          </button>
          <button
            aria-label="Go to the Next Month"
            class="rdp-button_next"
            type="button"
          >
            <svg
              class="rdp-chevron"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <polygon
                points="8 18.612 14.1888889 12.5 8 6.37733333 9.91111111 4.5 18 12.5 9.91111111 20.5"
              />
            </svg>
          </button>
        </nav>
        <div
          class="rdp-month"
        >
          <div
            class="rdp-month_caption"
          >
            <div
              class="rdp-dropdowns"
            >
              <span
                aria-live="polite"
                role="status"
              >
                noviembre
              </span>
              <span
                class="rdp-dropdown_root"
                data-disabled="false"
              >
                <select
                  aria-label="Choose the Year"
                  class="rdp-dropdown rdp-years_dropdown"
                >
                  <option
                    value="1925"
                  >
                    1925
                  </option>
                  <option
                    value="1926"
                  >
                    1926
                  </option>
                  <option
                    value="1927"
                  >
                    1927
                  </option>
                  <option
                    value="1928"
                  >
                    1928
                  </option>
                  <option
                    value="1929"
                  >
                    1929
                  </option>
                  <option
                    value="1930"
                  >
                    1930
                  </option>
                  <option
                    value="1931"
                  >
                    1931
                  </option>
                  <option
                    value="1932"
                  >
                    1932
                  </option>
                  <option
                    value="1933"
                  >
                    1933
                  </option>
                  <option
                    value="1934"
                  >
                    1934
                  </option>
                  <option
                    value="1935"
                  >
                    1935
                  </option>
                  <option
                    value="1936"
                  >
                    1936
                  </option>
                  <option
                    value="1937"
                  >
                    1937
                  </option>
                  <option
                    value="1938"
                  >
                    1938
                  </option>
                  <option
                    value="1939"
                  >
                    1939
                  </option>
                  <option
                    value="1940"
                  >
                    1940
                  </option>
                  <option
                    value="1941"
                  >
                    1941
                  </option>
                  <option
                    value="1942"
                  >
                    1942
                  </option>
                  <option
                    value="1943"
                  >
                    1943
                  </option>
                  <option
                    value="1944"
                  >
                    1944
                  </option>
                  <option
                    value="1945"
                  >
                    1945
                  </option>
                  <option
                    value="1946"
                  >
                    1946
                  </option>
                  <option
                    value="1947"
                  >
                    1947
                  </option>
                  <option
                    value="1948"
                  >
                    1948
                  </option>
                  <option
                    value="1949"
                  >
                    1949
                  </option>
                  <option
                    value="1950"
                  >
                    1950
                  </option>
                  <option
                    value="1951"
                  >
                    1951
                  </option>
                  <option
                    value="1952"
                  >
                    1952
                  </option>
                  <option
                    value="1953"
                  >
                    1953
                  </option>
                  <option
                    value="1954"
                  >
                    1954
                  </option>
                  <option
                    value="1955"
                  >
                    1955
                  </option>
                  <option
                    value="1956"
                  >
                    1956
                  </option>
                  <option
                    value="1957"
                  >
                    1957
                  </option>
                  <option
                    value="1958"
                  >
                    1958
                  </option>
                  <option
                    value="1959"
                  >
                    1959
                  </option>
                  <option
                    value="1960"
                  >
                    1960
                  </option>
                  <option
                    value="1961"
                  >
                    1961
                  </option>
                  <option
                    value="1962"
                  >
                    1962
                  </option>
                  <option
                    value="1963"
                  >
                    1963
                  </option>
                  <option
                    value="1964"
                  >
                    1964
                  </option>
                  <option
                    value="1965"
                  >
                    1965
                  </option>
                  <option
                    value="1966"
                  >
                    1966
                  </option>
                  <option
                    value="1967"
                  >
                    1967
                  </option>
                  <option
                    value="1968"
                  >
                    1968
                  </option>
                  <option
                    value="1969"
                  >
                    1969
                  </option>
                  <option
                    value="1970"
                  >
                    1970
                  </option>
                  <option
                    value="1971"
                  >
                    1971
                  </option>
                  <option
                    value="1972"
                  >
                    1972
                  </option>
                  <option
                    value="1973"
                  >
                    1973
                  </option>
                  <option
                    value="1974"
                  >
                    1974
                  </option>
                  <option
                    value="1975"
                  >
                    1975
                  </option>
                  <option
                    value="1976"
                  >
                    1976
                  </option>
                  <option
                    value="1977"
                  >
                    1977
                  </option>
                  <option
                    value="1978"
                  >
                    1978
                  </option>
                  <option
                    value="1979"
                  >
                    1979
                  </option>
                  <option
                    value="1980"
                  >
                    1980
                  </option>
                  <option
                    value="1981"
                  >
                    1981
                  </option>
                  <option
                    value="1982"
                  >
                    1982
                  </option>
                  <option
                    value="1983"
                  >
                    1983
                  </option>
                  <option
                    value="1984"
                  >
                    1984
                  </option>
                  <option
                    value="1985"
                  >
                    1985
                  </option>
                  <option
                    value="1986"
                  >
                    1986
                  </option>
                  <option
                    value="1987"
                  >
                    1987
                  </option>
                  <option
                    value="1988"
                  >
                    1988
                  </option>
                  <option
                    value="1989"
                  >
                    1989
                  </option>
                  <option
                    value="1990"
                  >
                    1990
                  </option>
                  <option
                    value="1991"
                  >
                    1991
                  </option>
                  <option
                    value="1992"
                  >
                    1992
                  </option>
                  <option
                    value="1993"
                  >
                    1993
                  </option>
                  <option
                    value="1994"
                  >
                    1994
                  </option>
                  <option
                    value="1995"
                  >
                    1995
                  </option>
                  <option
                    value="1996"
                  >
                    1996
                  </option>
                  <option
                    value="1997"
                  >
                    1997
                  </option>
                  <option
                    value="1998"
                  >
                    1998
                  </option>
                  <option
                    value="1999"
                  >
                    1999
                  </option>
                  <option
                    value="2000"
                  >
                    2000
                  </option>
                  <option
                    value="2001"
                  >
                    2001
                  </option>
                  <option
                    value="2002"
                  >
                    2002
                  </option>
                  <option
                    value="2003"
                  >
                    2003
                  </option>
                  <option
                    value="2004"
                  >
                    2004
                  </option>
                  <option
                    value="2005"
                  >
                    2005
                  </option>
                  <option
                    value="2006"
                  >
                    2006
                  </option>
                  <option
                    value="2007"
                  >
                    2007
                  </option>
                  <option
                    value="2008"
                  >
                    2008
                  </option>
                  <option
                    value="2009"
                  >
                    2009
                  </option>
                  <option
                    value="2010"
                  >
                    2010
                  </option>
                  <option
                    value="2011"
                  >
                    2011
                  </option>
                  <option
                    value="2012"
                  >
                    2012
                  </option>
                  <option
                    value="2013"
                  >
                    2013
                  </option>
                  <option
                    value="2014"
                  >
                    2014
                  </option>
                  <option
                    value="2015"
                  >
                    2015
                  </option>
                  <option
                    value="2016"
                  >
                    2016
                  </option>
                  <option
                    value="2017"
                  >
                    2017
                  </option>
                  <option
                    value="2018"
                  >
                    2018
                  </option>
                  <option
                    value="2019"
                  >
                    2019
                  </option>
                  <option
                    value="2020"
                  >
                    2020
                  </option>
                  <option
                    value="2021"
                  >
                    2021
                  </option>
                  <option
                    value="2022"
                  >
                    2022
                  </option>
                  <option
                    value="2023"
                  >
                    2023
                  </option>
                  <option
                    value="2024"
                  >
                    2024
                  </option>
                  <option
                    value="2025"
                  >
                    2025
                  </option>
                </select>
                <span
                  aria-hidden="true"
                  class="rdp-caption_label"
                >
                  2025
                  <svg
                    class="rdp-chevron"
                    height="18"
                    viewBox="0 0 24 24"
                    width="18"
                  >
                    <polygon
                      points="6.77 8 12.5 13.57 18.24 8 20 9.72 12.5 17 5 9.72"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
          <table
            aria-label="noviembre 2025"
            aria-multiselectable="false"
            class="rdp-month_grid"
            role="grid"
          >
            <thead>
              <tr
                class="rdp-weekdays"
              >
                <th
                  aria-label="lunes"
                  class="rdp-weekday"
                  scope="col"
                >
                  lu
                </th>
                <th
                  aria-label="martes"
                  class="rdp-weekday"
                  scope="col"
                >
                  ma
                </th>
                <th
                  aria-label="miércoles"
                  class="rdp-weekday"
                  scope="col"
                >
                  mi
                </th>
                <th
                  aria-label="jueves"
                  class="rdp-weekday"
                  scope="col"
                >
                  ju
                </th>
                <th
                  aria-label="viernes"
                  class="rdp-weekday"
                  scope="col"
                >
                  vi
                </th>
                <th
                  aria-label="sábado"
                  class="rdp-weekday"
                  scope="col"
                >
                  sá
                </th>
                <th
                  aria-label="domingo"
                  class="rdp-weekday"
                  scope="col"
                >
                  do
                </th>
              </tr>
            </thead>
            <tbody
              class="rdp-weeks"
            >
              <tr
                class="rdp-week"
              >
                <td
                  aria-hidden="true"
                  aria-label="lunes, 27 de octubre de 2025"
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2025-10-27"
                  data-hidden="true"
                  data-month="2025-10"
                  data-outside="true"
                >
                  27
                </td>
                <td
                  aria-hidden="true"
                  aria-label="martes, 28 de octubre de 2025"
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2025-10-28"
                  data-hidden="true"
                  data-month="2025-10"
                  data-outside="true"
                >
                  28
                </td>
                <td
                  aria-hidden="true"
                  aria-label="miércoles, 29 de octubre de 2025"
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2025-10-29"
                  data-hidden="true"
                  data-month="2025-10"
                  data-outside="true"
                >
                  29
                </td>
                <td
                  aria-hidden="true"
                  aria-label="jueves, 30 de octubre de 2025"
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2025-10-30"
                  data-hidden="true"
                  data-month="2025-10"
                  data-outside="true"
                >
                  30
                </td>
                <td
                  aria-hidden="true"
                  aria-label="viernes, 31 de octubre de 2025"
                  class="rdp-day rdp-hidden rdp-outside"
                  data-day="2025-10-31"
                  data-hidden="true"
                  data-month="2025-10"
                  data-outside="true"
                >
                  31
                </td>
                <td
                  aria-label="sábado, 1 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-01"
                >
                  1
                </td>
                <td
                  aria-label="domingo, 2 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-02"
                >
                  2
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  aria-label="lunes, 3 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-03"
                >
                  3
                </td>
                <td
                  aria-label="martes, 4 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-04"
                >
                  4
                </td>
                <td
                  aria-label="miércoles, 5 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-05"
                >
                  5
                </td>
                <td
                  aria-label="jueves, 6 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-06"
                >
                  6
                </td>
                <td
                  aria-label="viernes, 7 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-07"
                >
                  7
                </td>
                <td
                  aria-label="sábado, 8 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-08"
                >
                  8
                </td>
                <td
                  aria-label="domingo, 9 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-09"
                >
                  9
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  aria-label="lunes, 10 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-10"
                >
                  10
                </td>
                <td
                  aria-label="martes, 11 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-11"
                >
                  11
                </td>
                <td
                  aria-label="miércoles, 12 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-12"
                >
                  12
                </td>
                <td
                  aria-label="jueves, 13 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-13"
                >
                  13
                </td>
                <td
                  aria-label="viernes, 14 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-14"
                >
                  14
                </td>
                <td
                  aria-label="sábado, 15 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-15"
                >
                  15
                </td>
                <td
                  aria-label="domingo, 16 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-16"
                >
                  16
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  aria-label="lunes, 17 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-17"
                >
                  17
                </td>
                <td
                  aria-label="martes, 18 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-18"
                >
                  18
                </td>
                <td
                  aria-label="miércoles, 19 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-19"
                >
                  19
                </td>
                <td
                  aria-label="jueves, 20 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-20"
                >
                  20
                </td>
                <td
                  aria-label="viernes, 21 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-21"
                >
                  21
                </td>
                <td
                  aria-label="sábado, 22 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-22"
                >
                  22
                </td>
                <td
                  aria-label="domingo, 23 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-23"
                >
                  23
                </td>
              </tr>
              <tr
                class="rdp-week"
              >
                <td
                  aria-label="lunes, 24 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-24"
                >
                  24
                </td>
                <td
                  aria-label="Today, martes, 25 de noviembre de 2025"
                  class="rdp-day rdp-today"
                  data-day="2025-11-25"
                  data-today="true"
                >
                  25
                </td>
                <td
                  aria-label="miércoles, 26 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-26"
                >
                  26
                </td>
                <td
                  aria-label="jueves, 27 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-27"
                >
                  27
                </td>
                <td
                  aria-label="viernes, 28 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-28"
                >
                  28
                </td>
                <td
                  aria-label="sábado, 29 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-29"
                >
                  29
                </td>
                <td
                  aria-label="domingo, 30 de noviembre de 2025"
                  class="rdp-day"
                  data-day="2025-11-30"
                >
                  30
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</body>
`;
